Failure from t/15-clone-xs-objects.t - TRIAL 0.48_06

View more from https://magpie.cpantesters.org/results/00113336-10f5-11f1-9490-f34b6e8775ea

CI Failure reported for Perl 5.22 (does not always occur, probably config dependent)

```
    #   Failed test 'bstr() on cloned object does not die'
    #   at t/15-clone-xs-objects.t line 88.
    # bstr() died: failed to fetch mpz pointer at /opt/perl-5.22.2/lib/5.22.2/Math/BigInt.pm line 3347.
    # Looks like you failed 1 test of 5.

#   Failed test 'clone Math::BigInt::GMP objects (issue #16)'
#   at t/15-clone-xs-objects.t line 97.

    #   Failed test 'bstr() on cloned float does not die'
    #   at t/15-clone-xs-objects.t line 122.
    # bstr() died: failed to fetch mpz pointer at /opt/perl-5.22.2/lib/5.22.2/Math/BigFloat.pm line 3861.
    # Looks like you failed 1 test of 3.

#   Failed test 'clone Math::BigFloat::GMP objects (related)'
#   at t/15-clone-xs-objects.t line 125.
# Looks like you failed 2 tests of 4.
t/15-clone-xs-objects.t .. 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/4 subtests 
t/16-threads-shared.t .... skipped: Perl not compiled with thread support (useithreads)
t/17-threads-classdbi.t .. skipped: Perl not compiled with thread support (useithreads)
t/18-overload.t .......... ok
t/19-dualvar.t ........... ok
t/dclone.t ............... ok

Test Summary Report
-------------------
t/15-clone-xs-objects.t (Wstat: 512 Tests: 4 Failed: 2)
  Failed tests:  3-4
  Non-zero exit status: 2
Files=20, Tests=254,  1 wallclock secs ( 0.03 usr  0.04 sys +  0.58 cusr  0.17 csys =  0.82 CPU)
Result: FAIL
Failed 1/20 test programs. 2/254 subtests failed.
make: *** [test_dynamic] Error 255
```

--- a/Clone.xs
+++ b/Clone.xs
@@ -474,9 +474,11 @@ sv_clone (SV * ref, HV* hseen, int depth, int rdepth, AV * weakrefs)
 #if defined(MGf_DUP) && defined(sv_magicext)
           /* If the ext magic has a dup callback (e.g. Math::BigInt::GMP),
            * clone it properly via sv_magicext + svt_dup.
-           * Otherwise skip it (e.g. DBI handles have no dup). */
-          if (mg->mg_virtual && mg->mg_virtual->svt_dup
-              && (mg->mg_flags & MGf_DUP))
+           * Otherwise skip it (e.g. DBI handles have no dup).
+           * Note: we check only for svt_dup presence, not MGf_DUP flag,
+           * because some older XS modules (e.g. Math::BigInt::GMP on
+           * Perl 5.22) provide svt_dup without setting MGf_DUP. (GH #76) */
+          if (mg->mg_virtual && mg->mg_virtual->svt_dup)
           {
             MAGIC *new_mg;
             new_mg = sv_magicext(clone, mg->mg_obj,

